source/material_model/visco_plastic.cc

```diff
@@ -916,6 +954,45 @@ namespace aspect
                              "is not automatically used. Values of 100e6--1000e6 $Pa$ have been used "
                              "in previous models. Units: $Pa$");
 
+          // Transition from maximum spcrust viscosity to flow-law defined value
+          // over specified pressure range
+          prm.declare_entry ("Use fixed spcrust viscosity", "false", Patterns::Bool (),
+                             "Transition the viscosity of a compositional field called spcrust "
+                             "from a constant value to the value determined by the flow law parameters. "
+                             "Units: None");
+          prm.declare_entry ("Maximum spcrust viscosity", "1e28", Patterns::Double(0),
+                             "Maximum viscosity for the composition called spcrust. Using a value of 1e20 $Pa \\, s$"
+                             "would create a weak layer that smoothly increases to the viscosity "
+                             "determined by the flow law parameters (if these predict a higher "
+                             "value). Units: $Pa \\, s$");
+          prm.declare_entry ("Minimum transition pressure spcrust viscosity", "0.0", Patterns::Double(0),
+                             "Pressure at which to start the smooth transition from "
+                             "the maximum spcrust viscosity to the viscosity determined by"
+                             "the flow law. A value of 2.0e9 $Pa$ would correspond to a "
+                             "depth of about 60 km. Units: $Pa$");
+          prm.declare_entry ("Maximum transition pressure spcrust viscosity", "0.0", Patterns::Double(0),
+                             "Pressure at which to end smooth transition from "
+                             "the maximum spcrust viscosity to the viscosity determined by"
+                             "the flow law. A value of 3.9e9 $Pa$ would correspond to a "
+                             "depth of about 120 km. Units: $Pa$");
+
+          // Transition the spcrust density from defined value by a delta-rho
```

Comment:
What is the "defined value" for density? Is this just the regular density defined by the reference density, temperature and thermal expansivity?

Reply:
defined value is that given by the equation of state in the material model. I've changed this to say equation of stated defined value